cv32e40x/tests/programs/custom/illegal_instr_test/illegal_instr_test.S

```diff
@@ -47739,16 +47716,99 @@ main:
-fail:
-    li x18, 1
-test_end:
-    li x17, CV_VP_STATUS_FLAGS_BASE
-    sw x18,0(x17)
-    j _exit
+    addi    sp,sp,92
+    ret
+
+test_bitmanip:
+    add x3, x1, zero // backup return address
+    jal push_volatile_gpr_stack
+    .word(0x612ed793) // Zbb rori
+    .word(0x4088e233) // Zbb orn
+    .word(0x0a9e15b3) // Zbc clmul
+    .word(0x411f61b3) // Zbb orn
+    .word(0x0b8bbd33) // Zbc clmulh
+    .word(0x485b5733) // Zbs bext
+    .word(0x6174ddb3) // Zbb ror
+    .word(0x69e11d93) // Zbs binvi
+    .word(0x69041db3) // Zbs binv
+    .word(0x28ef9093) // Zbs bseti
+    .word(0x6108d693) // Zbb rori
+    .word(0x49031f13) // Zbs bclri
+    .word(0x29ab9db3) // Zbs bset
+    .word(0x6089d413) // Zbb rori
+    .word(0x0a202d33) // Zbc clmulr
+    .word(0x0a5393b3) // Zbc clmul
+    .word(0x48eb9293) // Zbs bclri
+    .word(0x48849613) // Zbs bclri
+    .word(0x60355893) // Zbb rori
+    .word(0x40fd7e33) // Zbb andn
+    .word(0x600b1793) // Zbb clz
+    .word(0x48085c33) // Zbs bext
+    .word(0x21b1c833) // Zba sh2add
+    .word(0x207fe233) // Zba sh3add
+    .word(0x48d59eb3) // Zbs bclr
+    .word(0x0a0c2a33) // Zbc clmulr
+    .word(0x29139693) // Zbs bseti
+    .word(0x492a9c93) // Zbs bclri
+    .word(0x40c3fdb3) // Zbb andn
+    .word(0x40756233) // Zbb orn
+    .word(0x492ad313) // Zbs bexti
+    .word(0x48535833) // Zbs bext
+    .word(0x214840b3) // Zba sh2add
+    .word(0x494cd493) // Zbs bexti
+    .word(0x219963b3) // Zba sh3add
+    .word(0x0af1b3b3) // Zbc clmulh
+    .word(0x40234433) // Zbb xnor
+    .word(0x20836cb3) // Zba sh3add
+    .word(0x0a099e33) // Zbc clmul
+    .word(0x20e86233) // Zba sh3add
+    .word(0x21e1e5b3) // Zba sh3add
+    .word(0x0ac23233) // Zbc clmulh
+    .word(0x20c0a333) // Zba sh1add
+    .word(0x49b5df33) // Zbs bext
+    .word(0x408b7533) // Zbb andn
+    .word(0x0a5e3933) // Zbc clmulh
+    .word(0x61b591b3) // Zbb rol
+    .word(0x0bad5d33) // Zbb minu
+    .word(0x40f5fcb3) // Zbb andn
+    .word(0x407342b3) // Zbb xnor
+    .word(0x203dc9b3) // Zba sh2add
+    .word(0x60e45633) // Zbb ror
+    .word(0x49305a93) // Zbs bexti
+    .word(0x0a1537b3) // Zbc clmulh
+    .word(0x0aa1ef33) // Zbb max
+    .word(0x692f92b3) // Zbs binv
+    .word(0x0b0d5e33) // Zbb minu
+    .word(0x0bd49a33) // Zbc clmul
+    .word(0x48da5233) // Zbs bext
+    .word(0x2177c1b3) // Zba sh2add
+    .word(0x61cb5c93) // Zbb rori
+    .word(0x21a2e433) // Zba sh3add
+    .word(0x41f8f8b3) // Zbb andn
+    .word(0x29ce18b3) // Zbs bset
+    .word(0x284a1193) // Zbs bseti
+    .word(0x0bffc233) // Zbb min
+    .word(0x0bda4533) // Zbb min
+    .word(0x0b59a333) // Zbc clmulr
+    .word(0x402e60b3) // Zbb orn
+    .word(0x2024abb3) // Zba sh1add
+    .word(0x0afe5db3) // Zbb minu
+    .word(0x409d63b3) // Zbb orn
+    .word(0x499cd093) // Zbs bexti
+    .word(0x0b646433) // Zbb max
+    .word(0x49a99d33) // Zbs bclr
+    .word(0x0a5eff33) // Zbb maxu
+    .word(0x68779c93) // Zbs binvi
+    li x16, 77
```

Comment:
Checks if the correct number of invalid instructions have been detected (either 0 or 77). If 77, then bitmanip is not detected and control is handed back to main. If detected, the value a55a5aa5 is stored to flag that we have valid bitmanip instructions. In case of valid instructions, the GPRs are restored to their initial state, since the random instructions above have modified their state.